src/firefly/js/ui/NaifidPanel.jsx

```diff
@@ -0,0 +1,200 @@
+/*
+ * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
+ */
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {get} from 'lodash';
+import {getFeedback, formatPosForTextField} from './TargetPanelWorker.js';
+import {resolveNaifidObj} from  './NaifidPanelWorker.js';
+import {fieldGroupConnector} from './FieldGroupConnector.jsx';
+import FieldGroupUtils from '../fieldGroup/FieldGroupUtils.js';
+import {dispatchActiveTarget, getActiveTarget} from '../core/AppDataCntlr.js';
+import {isValidPoint, parseWorldPt} from '../visualize/Point.js';
+import {SuggestBoxInputField} from './SuggestBoxInputField';
+import {flux} from 'firefly/Firefly.js';
+import {TargetFeedback} from './TargetFeedback';
+import {dispatchValueChange} from '../fieldGroup/FieldGroupCntlr';
+
+
+const LABEL_DEFAULT='Moving Target Name:';
+
+
+class NaifidPanelView extends PureComponent {
+
+    constructor(props) {
+        super(props);
+        this.state = {suggestions: '',fld: FieldGroupUtils.getGroupFields(this.props.groupKey)};
+        this.getSuggestions = this.getSuggestions.bind(this);
+    }
+
+    componentWillUnmount() {
+        if (this.removeListener) this.removeListener();
+        this.iAmMounted = false;
+    }
+
+    componentDidMount() {
+        this.iAmMounted = true;
+        this.unbinder = FieldGroupUtils.bindToStore(this.props.groupKey, (fields) => {
+            if (fields !== this.state && this.iAmMounted) {
+                this.setState({fields});
+            }
+        });
+    }
+
+
+    getSuggestions(val) {
+            const rval = resolveNaifidObj(val);
+            if (!rval.p) return undefined;
+            return rval.p.then((response)=>{
+                if(response.valid) {
+                    let suggestionsList = response.data || [];
+
+                    if(this.iAmMounted && this.state.suggestions !== suggestionsList) {
+                        this.setState({suggestions: suggestionsList});
+                    }else if(this.state.suggestions === suggestionsList){
+                        this.setState({suggestions:''});
```

Comment:
If you're checking for `this.iAmMounted` before `setState` on top, I am pretty sure you need to  check here too.

Reply:
Done, thanks!